sumy/nlp/tokenizers.py

```diff
@@ -77,6 +77,23 @@ def tokenize(self, text):
         return kkma.nouns(text)
 
 
+class GreekSentencesTokenizer:
+    """Calls sent_tokenize for greek, which doesn't split sentences on semicolon ';' 
+    or the greek question mark ';'. The regex below splits on both.
+    """
+    @classmethod
+    def tokenize(self, text):
+        sentences = nltk.sent_tokenize(text, language = 'greek')
+        sentences = [list(filter(None, re.split(r'(?<=[;;])\s+', sentence))) for sentence in sentences]
+        return [sentence.strip() for sent_list in sentences for sentence in sent_list]
+
+
+class GreekWordTokenizer:
+    @classmethod
+    def tokenize(self, text):
+        return list(filter(None, re.split("[ ,;;.\-!?:]+", text)))
```

Comment:
Again here, why are your `;` always doubled? 😄 And the backslash has to be escaped, but I am not sure if it's needed if you list `-` at the end.
```suggestion
        return list(filter(None, re.split("[ ,;.!?:-]+", text)))
```

Reply:
Greetings,

First of all thank you for all this valuable feedback. I am going to implement your suggested changes in the following days.

I am posting this as to solve your questions, with the help of the following list:

1.  The Greek stemmer essentially functions as a big lookup table for stems, due to the huge amount of suffixes and exceptions in the language. If you supply a word and a POS tag, and if it doesn't found in its table, it throws an exception, which I don't like as a design, but there isn't an alternative implementation for this really.
I'll try to catch a top level exception, if there are too many I'll use `except Exception`.

2. The code written in the `stem_word()` function is written by me and not copy pasted from any library.
I am terribly sorry for the miscommunication in the comment I was referring to the import statement `from greek_stemmer import stemmer as gr_stemmer`. Since its an external `import` of the library and I don't include any of its source code directly I'll delete this comment.

3. The sentence regex, splits sentences one or more times on the two distinct `;` Unicode characters (more details on those in item 4.). Why is this regex needed? Because `nltk_tokenize()` despite claiming to split Greek sentences, it doesn't split them on this character ';', which is the Greek question mark. This is definitely a bug.

4. Why using double question mark characters? Well they are not doubled or duplicated; the first one is the English semicolon [https://unicodelookup.com/#;/1](https://unicodelookup.com/#;/1), while the other is the Greek question mark [https://unicodelookup.com/#greek](https://unicodelookup.com/#greek). But why splitting on both of them would be another question? From my experience, as a Greek NLP practitioner, I have seen Greek texts using both characters in text, probably due to a mis-configuration or mis-use between the language packages in the operating system.

Thank you for your time!

Reply:
1. Thank you :)
2. Ok, that's good. Remove the GPL license file too then. I believe it's fine to use optional GPL license but the library has the license file if you install it so it's no needed to have it here. It's confusing IMHO. I know licensing is hard and I am not a lawyer but here it should be fine just use the library because it's optional dependency installed only by people who need it on their local computer and not part of the project.
3. Thanks, please explain it in the comment somewhere nearby to the RegExes please, maybe with links to https://unicode-table.com/en/003B/ and https://unicode-table.com/en/037E/ because I am sure I forget about it in the future 😄 
4. If this is true, you could probably report a bug into NLTK. I would do it myself, but if you have a bit time it would be probably better from you because, well, you know Greek language and I have no clue about it 😄